### ISSUE TITLE
Publish for Scala 3 (aka Dotty)

### ISSUE THREAD

**Issue by phdoerfler** [circe/circe-generic-extras#168]
I imagine this is easier said than done but, well, now there's an issue for it.
Thanks!

**Comment by javierarrieta** [comment]
Referencing a question in the global circe Scala 3 plans:

https://github.com/circe/circe/issues/1754#issuecomment-851864122

**Comment by felixbr** [comment]
I took a look at the code and it's full of macros, shapeless 2, and other things which are not supported by Scala 3 at all.

My intention was to write a Scala 3 version from scratch with the same apis but no Scala 2 compatibility otherwise. In the meantime @Lasering started to [add configuration options to the Scala 3 version of `circe-generic`](https://github.com/circe/circe/pull/1800), so that will probably solve the problem for me.

Not sure which features will still be missing if the PR mentioned above is included in 0.15.0.

**Comment by grzegorz-bielski** [comment]
For the Scala 3 a lot of functionality (everything?) of `circe-generic-extras` is already available in `core` ([see](https://github.com/circe/circe/blob/series/0.14.x/modules/core/shared/src/main/scala-3/io/circe/derivation/Configuration.scala)).

However, for projects that need to be cross-compiled, it'd still be useful to have those functionalities provided using the `circe-generic-extras` [apis](https://github.com/circe/circe-generic-extras/blob/main/generic-extras/src/main/scala/io/circe/generic/extras/semiauto.scala) using some compat-layer.

**Comment by felixbr** [comment]
@grzegorz-bielski I agree that it would be useful. It's even more relevant for shapeless, which also hasn't provided a compatibility layer despite being so fundamental.

Sadly it's a decent amount of work to split everything and have essentially two variants of the library going forward (Scala2 + shapeless but also Scala3 without shapeless).

I have already migrated our work projects to Scala3 without circe-generic-extras, so I personally won't work on this.

**Comment by grzegorz-bielski** [comment]
@felixbr Actually I already did some work in that [direction](https://github.com/grzegorz-bielski/circe-generic-extras/tree/scala3-compat-layer) and it looks promising. 
I’ll try to open a PR in near future.
